Murcielago/Murcielago/UserInterface/Views/PatternViews/GridPatternView.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  GridPatternView.swift
+//  Murcielago
+//
+//  Created by Sam Franusic on 10/22/24.
+//
+
+import SwiftUI
+
+struct GridPatternView: View {
+    @StateObject var model = GridModel()
+    var body: some View {
+        VStack {
+            ForEach(0..<model.rows, id: \.self) { row in
+                HStack {
+                    ForEach(0..<model.columns, id: \.self) { column in
+                        Rectangle()
+                            .fill(
+                                model.state[row][column]
+                                ? .murcielagoPrimary
+                                : .clear
+                            )
+                            .aspectRatio(contentMode: .fit)
+
+                    }
+                }
+            }
+        }
+        .padding()
+        .onAppear {
+            model.turnOn()
+        }
+        .onTapGesture {
+            model.turnOn()
+        }
+    }
+}
+
+@MainActor
+class GridModel: ObservableObject {
+    let rows = 4
+    let columns = 4
+    @Published private(set) var state: [[Bool]]
+
+    init() {
+        state = Array(repeating: Array(repeating: false, count: columns), count: rows)
+    }
+
+    func turnOn() {
```

Comment:
Claude suggests that this could be simplified as follows:

```swift
func turnOn() async {
    // Corner animation
    await Task.sleep(for: .seconds(1))
    await withAnimation(.linear(duration: 2)) {
        turnOnCorners()
    }
    
    // Center animation
    await Task.sleep(for: .seconds(1))
    await withAnimation(.linear(duration: 2)) {
        turnOnCenter()
    }
    
    // All cells animation
    await Task.sleep(for: .seconds(1))
    await withAnimation(.linear(duration: 2)) {
        turnOnAll()
    }
    
    // Reset
    await Task.sleep(for: .seconds(2))
    turnAllOff()
}
```

I'm not sure if this is equivalent but it looks nicer to me.

Reply:
Done. Needed to add `try` to Task.sleep, and renamed the method also.